login-workflow/example/src/app/pages/pre-auth/pre-auth.component.ts

```diff
@@ -1,15 +1,43 @@
 import { Component } from '@angular/core';
 import { Router } from '@angular/router';
-import { AUTH_ROUTES } from '@pxblue/angular-auth-workflow';
+import { AUTH_ROUTES, PxbAuthSecurityService, SecurityContext } from '@pxblue/angular-auth-workflow';
 
 @Component({
     selector: 'app-pre-auth',
     template: `
-        <button color="primary" (click)="router.navigate([routes.AUTH_WORKFLOW])" mat-stroked-button>Go Login</button>
-        <button color="primary" (click)="router.navigate(['/dashboard'])" mat-flat-button>Go Auth Guarded Page</button>
+        isAuthenticated?: {{ isAuth }}
+        <br />
+        <button color="primary" (click)="goLogin()" mat-stroked-button>Go Login</button>
+        <button color="primary" (click)="goHome()" mat-stroked-button>Go Home</button>
+        <button color="primary" (click)="goDashboard()" mat-flat-button>Go Dashboard</button>
     `,
 })
 export class PreAuthComponent {
+    isAuth = false;
     routes = AUTH_ROUTES;
-    constructor(public router: Router) {}
+
+    constructor(public router: Router, public pxbSecurityService: PxbAuthSecurityService) {
+        this._listenForAuthStateChanges();
+    }
+
+    goLogin(): void {
+        this.routes.ON_AUTHENTICATED = '';
```

Comment:
The router doesn't infer anything; our pxbSecurityService attempts to infer the ON_AUTHENTICATED route  on app load, but it's not *that* smart.

Imagine you've loaded the app and told the security service to do its best to auto-direct me to the correct route post-login.  

If you load the app on the pre-auth page, then the ON_AUTHENTICATED route = pre-auth because that's the route the user intended to go to on app load.   If you're not logged in, and you click a button to go to the home/dashboard page, the auth guard will reroute to the login.  Post-login you are redirected back to the pre-auth.

Reply:
```
goLogin(): void {
      this.routes.ON_AUTHENTICATED = '';
      void this.router.navigate([this.routes.AUTH_WORKFLOW]);
  }
```

The ON_AUTHENTICATED route is being completely overwritten here on these button clicks, users can change this route whenever they need to.

Reply:
The ON_AUTHENTICATED route is being completely overwritten here on these button clicks, users can change this route whenever they need to.  


The router doesn't infer anything; our pxbSecurityService attempts to infer the ON_AUTHENTICATED route  on app load, but it's not *that* smart.

Imagine you've loaded the app and told the security service to do its best to auto-direct me to the correct route post-login.  

If you load the app on the pre-auth page, then the ON_AUTHENTICATED route = pre-auth because that's the route the user intended to go to on app load.   If you're not logged in, and you click a button to go to the home/dashboard page, the auth guard will reroute to the login.  Post-login you are redirected back to the pre-auth.

Reply:
```
goLogin(): void {
      this.routes.ON_AUTHENTICATED = '';
      void this.router.navigate([this.routes.AUTH_WORKFLOW]);
  }
```

Reply:
I gotcha Not ideal, but I think I understand the limitation. On the react side, when a user tries to navigate to a protected route we save their intended destination in the location / history object and send them back there once they are authenticated:

Save the destination:
https://github.com/pxblue/react-workflows/blob/dev/login-workflow/src/components/PrivateRoute.tsx#L23
Restore the destination:
https://github.com/pxblue/react-workflows/blob/a0d85f3cf636780ca45a1b9848cb615ec2328394/login-workflow/src/screens/PreAuthContainer.tsx#L33

Reply:
I'm actually not 100% sure what would happen if you were starting on a non-protected route though. Looking at the second link where we restore the destination, the default is the empty route too. But I'm not sure if the 'from' field would be populated by that initial public route. I guess we'll find out when we do this for the React side :)

Reply:
I prefer manually setting the ON_AUTH route before navigating here because it gives the adopters more control over what's happening.  I don't like the idea of the ON_AUTHENTICATED route being updated each time a failed navigation occurs; I like the flexibility of giving the users to set it themselves without worrying about it being updated by our module.  That's part of the reason why I made the `inferRoute` logic an optional method call, because there was confusion on where/when the ON_AUTH route was being set.

Reply:
I'm goin to merge this in now.